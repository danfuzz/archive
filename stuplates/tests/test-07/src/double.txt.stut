== true
${ (:double: 0.0 == 0.0) }
${ (:double: 0.0 == -0.0) }
${ (:double: 10.2 == 10.2) }

== false
${ (:double: 0.1 == 1.0) }
${ (:double: 100.2 == -100.2) }

!= true
${ (:double: 0.0 != 1.2) }
${ (:double: 100.0 != -100.15) }

!= false
${ (:double: 0.0 != 0.0) }
${ (:double: 10.5 != 10.5) }

< true
${ (:double: 5.1 < 6.2) }
${ (:double: 18.123 < 3600.321) }

< false
${ (:double: 5.555 < -5.555) }
${ (:double: -10.10 < -20.20) }

> true
${ (:double: .2 > -.2) }
${ (:double: 100.4 > 2.66) }

> false
${ (:double: 4.9 > 5.99) }
${ (:double: -1.01 > 10.01) }

<= true
${ (:double: 0.0 <= 0.0) }
${ (:double: 0.3 <= 10.67) }

<= false
${ (:double: 0.9 <= -1.1) }
${ (:double: 5.0 <= 2.0) }

>= true
${ (:double: 20.3 >= 20.3) }
${ (:double: -20.1 >= -22.2) }

>= false
${ (:double: 1.1 >= 5.1) }
${ (:double: 0.0 >= 50.2) }

+
${ (:double: 0.0 + 0.0) } 0.0
${ (:double: 1.0 + 3.2) } 4.2
${ (:double: +"05") } 5.0
${ (:double: +"-20.4") } -20.4
${ (:double: +5) } 5.0
${ def x := -5; (:double: +x) } -5.0
${ (:double: +true) } 1.0
${ (:double: +false) } 0.0
${ (:double: +"nAN") } NaN
${ (:double: +"-infINity") } -Infinity
${ (:double: +"InfinitY") } Infinity

-
${ (:double: 0.0 - 0.0) } 0.0
${ (:double: 2.5 - 5.5) } -3.0
${ (:double: -3.4) } -3.4
${ (:double: -0.0) } -0.0

*
${ (:double: 33.1 * 2.0) } 66.2
${ (:double: 66.6 * 0.0) } 0.0
${ (:double: 32.0 * -2.0) } -64.0

/
${ (:double: 15.0 / 2.0) } 7.5
${ (:double: 16.0 / 2.0) } 8.0
${ (:double: -8.55 / 8.55) } -1.0

%
${ (:double: 15.0 % 2.0) } 1.0
${ (:double: 16.0 % 2.0) } 0.0
${ (:double: -800.0 % 800.0) } 0.0
${ (:double: -15.0 % 2.0) } -1.0
${ (:double: -15.0 % -2.0) } -1.0

%%
${ (:double: 15.0 %% 2.0) } 1.0
${ (:double: 16.0 %% 2.0) } 0.0
${ (:double: -8.0 %% 8.0) } 0.0
${ (:double: -15.0 %% 2.0) } 1.0

**
${ (:double: 1.0 ** 10.0) } 1.0
${ (:double: 2.0 ** 8.0) } 256.0
${ (:double: 2.0 ** -1.0) } 0.5
${ (:double: 2.0 ** -2.0) } 0.25
${ (:double: 4.0 ** 0.5) } 2.0
