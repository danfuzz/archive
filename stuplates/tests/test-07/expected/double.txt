== true
true
true
true

== false
false
false

!= true
true
true

!= false
false
false

< true
true
true

< false
false
false

> true
true
true

> false
false
false

<= true
true
true

<= false
false
false

>= true
true
true

>= false
false
false

+
0.0 0.0
4.2 4.2
5.0 5.0
-20.4 -20.4
5.0 5.0
-5.0 -5.0
1.0 1.0
0.0 0.0
NaN NaN
-Infinity -Infinity
Infinity Infinity

-
0.0 0.0
-3.0 -3.0
-3.4 -3.4
-0.0 -0.0

*
66.2 66.2
0.0 0.0
-64.0 -64.0

/
7.5 7.5
8.0 8.0
-1.0 -1.0

%
1.0 1.0
0.0 0.0
-0.0 0.0
-1.0 -1.0
-1.0 -1.0

%%
1.0 1.0
0.0 0.0
-0.0 0.0
1.0 1.0

**
1.0 1.0
256.0 256.0
0.5 0.5
0.25 0.25
2.0 2.0
